substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -437,4 +437,16 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::AllowedRenewalDropped { core, when });
 		Ok(())
 	}
+
+	/// If there is a sale ongoing returns the current price of a core.
+	pub fn api_sale_price() -> Option<BalanceOf<T>> {
+		let status = Status::<T>::get()?;
+		let sale = SaleInfo::<T>::get()?;
+
+		if sale.first_core < status.core_count || sale.cores_sold < sale.cores_offered {
```

Comment:
Can you explain the second part of the `or`, why don't we have a sale price if cores_sold < cores_offered? Also:

1. This looks like code duplication that could get out of sync. If we don't have a price for some reason, this should not be determined here, but e.g. in something like `sale_price`.
2. Wouldn't a reason for not having a sale price be of interest? E.g. would it make sense to make the return type more expressive?

Reply:
> Can you explain the second part of the or

Yes, this was incorrect the comparison sign needs to be flipped for actually both parts of the `or`.

> this should not be determined here, but e.g. in something like sale_price.

I would rather leave `sale_price` as a low level function without much contextual information and checks. I extracted the checks in a new function.

> Wouldn't a reason for not having a sale price be of interest? E.g. would it make sense to make the return type more expressive?

We can return more information by returning an error instead of `None`. I didn't do it since it is a bit weird to return a `DispatchError` from a runtime api function, but it might be worth it for more expressiveness.